code/game/objects/items/control_wand.dm

```diff
@@ -155,7 +155,71 @@
 	busy = FALSE
 	icon_state = "hacktool"
 
+/// How long before you can "jangle" your keyring again (to prevent spam)
+#define JANGLE_COOLDOWN 10 SECONDS 
+
+/obj/item/door_remote/janikeyring
+	name = "janitor's keyring"
+	desc = "An absolutely unwieldy set of keys attached to a metal ring."
+	icon_state = "keyring"
+	item_state = "keyring"
+	var/busy = FALSE
+	var/cooldown = 0
+	var/hack_speed
```

Comment:
Define in the proc below maybe

Reply:
`hack_speed` is only really set when used, and is dependent on the role of the user so that non-Janitors take WAY longer. Unsure if there's a nice way to define it in the proc without just copy-pasting what's already done below.